### PR TITLE
GLTFLoader: Implement KHR_texture_basisu support

### PR DESCRIPTION
**Pull request by zeux**
This change implements support for (draft) KHR_texture_basisu extension
as described in https://github.com/KhronosGroup/glTF/pull/1751.

To make this work, two general changes to GLTFLoader were necessary:

- The plugin extension framework now supports overriding loadTexture
- To simplify implementation, the bulk of the loading functionality was
extracted into loadTextureImage.

As with other dependent loaders, it's the caller's responsibility to
call setKTX2Loader on the GLTFLoader before loading files with this
extension.

### CODE REVIEW

**Comment by zeux** [comment]
Please note that the last commit in this PR slightly changes the behavior for loading embedded images in that they no longer call LoadingManager.getHandler. This seems reasonable to me since the path passed to getHandler in these cases is something like

    "blob:http://127.0.0.1:8000/981682e7-df36-4ab5-bcbe-5fe65f7bc59d"

So it's hard to imagine how to match this via regexps in a reasonable manner, but if this is not desireable then I'm happy to revert this part.

**Comment by WestLangley** [comment]
Friendly reminder: These changes will get overwritten with `node utils/modularize.js`. For now, you will have to edit the `js/` file, instead. Then run `modularize.js`.

**Comment by zeux** [comment]
Thanks! I didn't realize that the pre-module files are still maintained since KTX2Loader is only available as a jsm.

Also added an example (live link https://raw.githack.com/zeux/three.js/gltf-ktx2/examples/webgl_loader_gltf_ktx2.html) for this change, again let me know if it's better to remove that from the PR (the model is from glTF-Sample-Models repository). It looks like to actually commit this I'll need to generate a screenshot though, unsure how to do that...

**Comment by WestLangley** [comment]
>I'll need to generate a screenshot though, unsure how to do that...

https://github.com/mrdoob/three.js/tree/dev/test/e2e

    npm run make-screenshot <example1_name>  // omit the .html extension

I think you can wait until you get further feedback, though. :-)

**Comment by donmccurdy** [comment]
> I didn't realize that the pre-module files are still maintained since KTX2Loader is only available as a jsm.

We'll be maintaining the `js/` directory until the end of 2020, but we're only adding new files (like KTX2Loader) to `jsm/` in the meantime.

***

The KTX 2.0 specification is stable and very unlikely to change further before ratification, so it's reasonable to release KTX2Loader with r119 and encourage developers to use it, once bugs like #19793 (and perhaps a few more things) are resolved.

The `KHR_texture_basisu` extension for glTF is *not* stable yet, and in fact I do expect it to change somewhat, so I'm a bit torn about adding it to GLTFLoader now. Since there are already users putting `.dds` and `.basis` textures into glTF files unofficially — which are definitely not going to be official glTF extensions — maybe we should go ahead and support `.ktx2` in GLTFLoader to move people towards the long-term direction, even if it isn't stable yet?

That said, I think I'd rather not include an example yet — I don't want to draw that much attention to it until later.

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -535,6 +550,46 @@ THREE.GLTFLoader = ( function () {
...
+	/**
+	 * BasisU Texture Extension
+	 *
+	 * Specification: https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_texture_basisu
+	 * (Link above TBD, currently in PR https://github.com/KhronosGroup/glTF/pull/1751)
+	 */
+	function GLTFTextureBasisU( parser ) {
+
+		this.parser = parser;
+		this.name = "KHR_texture_basisu";
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js:562]: Could you add this to the `EXTENSIONS` enum, instead?

**Comment by zeux** [comment]
> That said, I think I'd rather not include an example yet — I don't want to draw that much attention to it until later.

That makes sense - I can remove the example for now. I think it would be good to have support for this otherwise, Babylon.js has it as well - even if there are some changes in the extension later presumably they wouldn't be as drastic as to require a completely different implementation.

File: examples/js/loaders/GLTFLoader.js (hunk 1 above)
Comment (zeux) [reply examples/js/loaders/GLTFLoader.js]: Done

**Review by donmccurdy — APPROVED** [review]
Thank you! Extension is working nicely here.  👍

**Comment by donmccurdy** [comment]
Added this to the experimental version of my viewer, https://gltf-viewer-experimental.donmccurdy.com.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**